qldpc/codes/classical.py

```diff
@@ -238,3 +238,37 @@ def as_directed_subgraph(subgraph: nx.Graph) -> nx.DiGraph:
                 directed_subgraph[node_a][edge]["sort"] = sort_data[node_a]
                 directed_subgraph[node_b][edge]["sort"] = sort_data[node_b]
         return directed_subgraph
+
+class SimplexCodes(ClassicalCode):
+    """Classical Simplex codes: used to generate the SHYPS (type of Subsystem
+        Hypergraph product code) codes described in https://arxiv.org/pdf/2502.07150"""
+
+    def __init__(self, r:int)->None:
+        self._exact_distance = 2**(r-1)
+        self._num_bits = 2**r-1
+        self._field = galois.GF(2)
+
+        self.generating_poly = SimplexCodes.get_generating_poly(r)
+        
+        ## define parity check matrix using generating polynomial
+        matrix = self.field.Zeros((self._num_bits, self._num_bits))
+        for i in range(self._num_bits):
+            for j in self.generating_poly:
+                matrix[i,(j+i)%self._num_bits] = 1
+
+        ClassicalCode.__init__(self, matrix)
+
+    @staticmethod
+    def get_generating_poly(r:int) -> npt.NDArray[np.int_]: 
```

Comment:
This doesn't return a "polynomial" per se.  Maybe rename to something like `get_generating_polynomial_exponents`?

Also, no need for a numpy array; this method can return, say, a `list[int]` or `tuple[int, ...]`.